src/lib/protocol_version/protocol_version.ml

```diff
@@ -71,6 +71,104 @@ module Make_str (A : Wire_types.Concrete) = struct
      negative numbers
   *)
   let is_valid t = t.transaction >= 1 && t.network >= 1 && t.patch >= 0
+
+  let older_than_current t =
+    transaction t < transaction current
+    || (transaction t = transaction current && network t < network current)
+    || transaction t = transaction current
+       && network t = network current
+       && patch t < patch current
+
+  let gen =
+    let open Quickcheck.Generator.Let_syntax in
+    let%map transaction = Quickcheck.Generator.small_non_negative_int
```

Comment:
can you use `small_positive_int` for transaction and network, then you don't have to add 1